### PR TITLE
ENT-3334 | SAP SF customer is seeing courses as not open for enrollment when they should be

### PR DESCRIPTION
**Pull request by HammadAhmadWaqas**
comparing `status` of the course_run with the `unpublished` instead of 'published' to handle cases where unpublished key is missing.

**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [ ] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [ ] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [Travis](https://travis-ci.org/github/edx/edx-enterprise), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

File: enterprise/utils.py [hunk 1]
```diff
@@ -822,12 +822,12 @@ def is_course_run_upgradeable(course_run):
...
 
-def is_course_run_published(course_run):
+def is_course_run_unpublished(course_run):
     """
-    Return True if the course run's status value is "published".
+    Return True if the course run's status value is "unpublished".
     """
     if course_run:
-        if course_run.get('status') == 'published':
+        if course_run.get('status') == 'unpublished':
```
Comment (hasnain-naveed) [inline enterprise/utils.py:830]: What are the other statuses?

Comment (saleem-latif) [inline enterprise/utils.py:830]: Why was this causing the issue? Do we not have the correct value of status for published courses?

Comment (HammadAhmadWaqas) [reply enterprise/utils.py:830]: @saleem-latif 
the main purpose of this method was to filter out all unpublished course_runs from the enrollable courses.
it was doing it successfully for all the course_runs where the status key was present, but the course_runs where the status key was missing(as it's not mandatory) this check was returning incorrect value. (None == published)
So just modified this method in a way to handle scenarios where the `status` key is not present to consider that course available for enrollment.
@saleem-latif is it helping to answer your question?

Comment (HammadAhmadWaqas) [reply enterprise/utils.py:830]: this was the PR when we wrote this method.
https://github.com/edx/edx-enterprise/pull/795

Comment (saleem-latif) [reply enterprise/utils.py:830]: So for missing status, the course will be considered published, right?

**Comment by HammadAhmadWaqas** [comment]
Yes

**Review by saleem-latif — APPROVED** [review]

**Review by hasnain-naveed — APPROVED** [review]

**Comment by HammadAhmadWaqas** [comment]
closing this PR, in the favor of other solution.
https://github.com/edx/course-discovery/pull/2758

**Comment by mattdrayer** [comment]
Yay, I didn't break it after all :)

**Pull request closed without merging**